module_test.go

```diff
@@ -76,6 +76,42 @@ func TestModuleSuccess(t *testing.T) {
 		defer app.RequireStart().RequireStop()
 	})
 
+	t.Run("provide a value to a soft group value from nested modules", func(t *testing.T) {
+		t.Parallel()
+		type Param struct {
+			fx.In
+
+			Foos []string `group:"foo,soft"`
+			Bar  int
+		}
+		type Result struct {
+			fx.Out
+
+			Foo string `group:"foo"`
```

Comment:
can you add an error case where `soft` is specified to an fx.Out struct? Also one where it's being speified using fx.Annotate?

i.e.
`fx.Annotate(someFunc, fx.ResultTags(`group:"something,soft"`))`
should error out. 
I believe that won't error out currently, because you haven't changed the implementation.

Take a look at annotated.go and look at the `apply` method for `resultTagsAnnotation`. You probably want to add something there 🙂